server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -452,6 +452,19 @@ public boolean isCompatible(Version version) {
         return compatible;
     }
 
+    /**
+     * Get the released version that is before the given version
+     */
+    public Version getPreviousVersion() {
+        for (int i = DeclaredVersionsHolder.DECLARED_VERSIONS.size() - 1; i >= 0; i--) {
+            final Version version = DeclaredVersionsHolder.DECLARED_VERSIONS.get(i);
+            if (version.isRelease() && version.before(this)) {
+                return version;
+            }
+        }
+        throw new IllegalArgumentException("cannot find a released version before [" + this + "]");
+    }
```

Comment:
This is `VersionUtils.getPreviousVersion`

Reply:
Yes basically copy/pasted the code from `VersionUtils.getPreviousVersion` which is only available in test. But it does not matter now. I am going to revert these changes as mentioned above.